openduck-py/openduck_py/routers/voice.py

```diff
@@ -168,20 +168,25 @@ def _inference(sentence: str):
 
             transcription = await loop.run_in_executor(None, _transcribe, audio_data)
             print("TRANSCRIPTION: ", transcription)
-
-            if not transcription:
-                return
-
-            t0 = time()
-
-            transcription = await loop.run_in_executor(None, _transcribe, audio_data)
-            print("transcription", transcription)
             await log_event(
                 db, self.session_id, "transcribed_audio", meta={"text": transcription}
             )
+            classify_prompt = prompt("intent-classification")
+            classification_response = await generate(
```

Comment:
Yes.

1) I think theres a chance this is not going to be very useful since ESP32 boards often have on device word recognition.
2) We could launch the API calls to respond and classify in parallel, or try to make the conversation continuation call output stop if detected.